tests/blockchain/eth/contracts/integration/test_intercontract_integration.py

```diff
@@ -913,7 +921,17 @@ def test_all(testerchain, token, escrow, policy_manager, adjudicator, user_escro
         testerchain.wait_for_receipt(tx)
         testerchain.time_travel(hours=1)
 
+    # Can't unlock re-stake parameter yet
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = user_escrow_proxy_1.functions.setReStake(False).transact({'from': ursula3})
+        testerchain.wait_for_receipt(tx)
+
     testerchain.time_travel(hours=1)
+    # Now can turn off re-stake
```

Comment:
(continuing the above) And this can be turned off the *same* period as the tokens are unlocked?

Reply:
These dates are not directly related, `lockReStakeUntilPeriod` matters for `reStake` and stake parameters are independent.
Only in `mint` method they are used together - if `reStake` is true then reward is added to sub stake